### ISSUE TITLE
INSERT .. RETURNING emulations shouldn't acquire multiple connections from ConnectionProvider when using ThreadLocalTransactionProvider

### ISSUE THREAD

**Issue by vptp** [jOOQ/jOOQ#13563]
### Expected behavior
Using SQLite, `ThreadLocalTransactionProvider` and a connection pool of size 1
`dao.insert(Pojo(null, ...))` should insert 1 new record into the database

### Actual behavior
Exception: "Connection is not available, request timed out"
The insert code attempts to get a second connection from the pool while the first connection has yet to be returned to the pool.

### Steps to reproduce the problem
MCVE: https://github.com/vptp/jOOQ-mcve Test case is in Kotlin
```
git clone https://github.com/vptp/jOOQ-mcve
cd jOOQ-mcve
mvn verify
```

### Workaround
Wrap all standalone `dao.insert(...)` statements in transactions.

### Notes
The insert code for SQLite performs two statements on the database, one to insert the row, then a second to fetch the ID of the newly inserted row.
When performing an insert outside of a transaction, this appears to cause two separate calls to `acquire()` a connection.
I would expect that all statements that are part of the insert to be using the same connection, triggering only a single call to `acquire()` a connection.

### Use case
Multi-thread application performing a mix of transactional and non-transactional statements on a SQLite database.
Single connection thread pool ensures that database access is single threaded. For this use case it is acceptable that other threads wait for the connection to be free.

### Versions

- jOOQ: 3.16.3
- Java: 11.0.15
- Database (include vendor): SQLite 3.28.0
- OS: Linux

**Comment by lukaseder** [comment]
I see now, so this is really specific to using the `ThreadLocalTransactionProvider`, but running queries in a non-transactional context. Clearly a bug in the `ThreadLocalTransactionProvider` semantics.

In that sense, a packport is certainly possible. Will look into the fix right away.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.7 (https://github.com/jOOQ/jOOQ/issues/13567), and 3.15.11 (https://github.com/jOOQ/jOOQ/issues/13568)

Thanks again for your report!
